### PR TITLE
[ALLUXIO-1739] Move Dataserver to Alluxio Worker

### PR DESCRIPTION
**Pull request by calvinjia**
https://tachyon.atlassian.net/browse/ALLUXIO-1739

This is a step to implementing ALLUXIO-1739. After moving the data server out of the block worker, we can share the data server between different types of workers.

### CODE REVIEW

File: core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java [hunk 1]
```diff
@@ -105,16 +103,6 @@ public void before() throws IOException {
...
 
   /**
-   * Stop the DataServer to clean up.
-   *
-   * @throws IOException if clean up fails
-   */
-  @After
-  public void after() throws IOException {
-    ((DataServer) Whitebox.getInternalState(mBlockWorker, "mDataServer")).close();
-    WorkerContext.reset();
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java]: We still need the WorkerContext reset due to
```
conf.set("alluxio.worker.tieredstore.level0.dirs.path",
         mFolder.newFolder().getAbsolutePath());
```
in the Before

**Comment by aaudiber** [comment]
</review-1>

File: core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java (hunk 1 above)
Comment (calvinjia) [inline core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java]: Good call, thanks.

**Comment by gpang** [comment]
LGTM

**Pull request merged**